task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -67,7 +70,22 @@ def reconcile(self, task_config: KubernetesTaskConfig) -> None:
         pass
 
     def kill(self, task_id: str) -> bool:
-        pass
+        """
+        Delete a Pod by name.
+
+        This function will have task_processing stop tracking the killed task/Pod and
+        will return True if the Pod was successfully killed (and False otherwise)
+        """
+        with self._lock:
+            self.task_metadata = self.task_metadata.discard(task_id)
```

Comment:
we should probably only discard after we send request successfully and receive an event that confirms the deletion

Reply:
ah, I was trying to figure out the exact flow of how the mesos plugin handled this - that makes sense tho

Reply:
I guess I should also make sure that this delete call doesn't block too!

Reply:
I need to do some testing, but I think that the return value of a `delete_*` call is just if the k8s api successfully processed the request - not if the underlying object was deleted. 

so the only change that we'll want to make here is deferring the metadata deletion to when we have a Watch setup to get Pod events if I'm understanding thing correctly!